tests/lib/c_lib/src/main.c

```diff
@@ -558,6 +558,45 @@ void test_tolower_toupper(void)
 	zassert_equal(strcmp(lw, tolw), 0, "tolower error");
 }
 
+void test_strsep_do(char *str, char *sep, int tlen, char **toks, bool expect)
+{
+	int len = 0;
+	char *p, *tok, buf[64+1] = {0};
+
+	strncpy(buf, str, 64);
+	p = buf;
+
+	while (len < tlen && (tok = strsep(&p, sep)) != NULL) {
+		printk("tok: '%s' sep: '%s'\n", tok, toks[len]);
+		if (strcmp(tok, toks[len]) != 0) {
+			break;
+		}
+		len++;
+	}
+	if (expect) {
+		zassert_equal(len, tlen,
+			      "strsep error '%s' / '%s'", str, sep);
+	} else {
+		zassert_not_equal(len, tlen,
+				  "strsep error '%s' / '%s'", str, sep);
+	}
+}
+
+void test_strsep(void)
+{
+	const char **tc01 = { "1", "2", "3", "4", "5" };
+
+	test_strsep_do("1,2,3,4,5",           ",",  5, tc01, true);
+	test_strsep_do(",, 1 ,2  ,3   4,5  ", ", ", 5, tc01, true);
+	test_strsep_do("1,,,2 3,,,4 5",       ", ", 5, tc01, true);
+	test_strsep_do("1,2 3,,,4 5  ",       ", ", 5, tc01, true);
+	test_strsep_do("0,1,,,2 3,,,4 5",     ", ", 5, tc01, false);
+	test_strsep_do("1,,,2 3,,,4 5",       ",",  5, tc01, false);
+	test_strsep_do("A,,,2,3,,,4 5",       ",",  5, tc01, false);
+	test_strsep_do("1,,,2,3,,,",          ",",  5, tc01, false);
```

Comment:
With real `strsep` this produces 11 tokens (parenthesized value is the value of the pointer referenced by the first argument)
```
tok 1: '1' (0x7ffc1d0184bf)
tok 2: '' ((nil))
tok 3: '' (0x7ffc1d0184c0)
tok 4: '' (0x7ffc1d0184c1)
tok 5: '2' (0x7ffc1d0184c3)
tok 6: '' ((nil))
tok 7: '3' (0x7ffc1d0184c5)
tok 8: '' ((nil))
tok 9: '' (0x7ffc1d0184c6)
tok 10: '' (0x7ffc1d0184c7)
tok 11: '' ((nil))
```

Reply:
I will change this to strtok_r so the behavior remains same.